### PR TITLE
[NU-1149] Validate disabled nodes expressions

### PR DESCRIPTION
**Pull request by mslabek**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala [hunk 1]
```diff
@@ -74,28 +74,49 @@ class UIProcessValidator(
...
+            fragmentResolver.resolveFragments(canonical.withoutDisabledNodes, processingType)
+
+          resolvedScenarioWithoutDisabledNodes match {
+            case Invalid(e)                          => formatErrors(e)
+            case Valid(scenarioWithoutDisabledNodes) =>
+              // FIXME: Validation errors for fragment nodes are not properly handled by FE
+              // We add typing data from disabled nodes to have typing and suggestions for expressions in disabled nodes
+              val resultWithoutDisabledNodes = validateAndFormatResult(scenarioWithoutDisabledNodes)
+              resultWithoutDisabledNodes.copy(nodeResults =
+                resultWithoutDisabledNodes.nodeResults ++ validationResult.nodeResults
```
Comment (wrzontek) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala:112]:
```suggestion
                validationResult.nodeResults
```

resultWithoutDisabledNodes is (or should be) a subset of it anyway

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala [hunk 2]
```diff
@@ -74,28 +74,49 @@ class UIProcessValidator(
...
+
+    // TODO: handle types when fragment resolution fails
+    val validationResult = resolvedScenarioResult match {
+      case Invalid(fragmentResolutionErrors) => formatErrors(fragmentResolutionErrors)
+      case Valid(scenario) =>
+        val validationResult = validateAndFormatResult(scenario)
+
+        val containsDisabledNodes =
+          canonical.withoutDisabledNodes.collectAllNodes.size != canonical.collectAllNodes.size
+        if (containsDisabledNodes) {
```
Comment (wrzontek) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala:100]: weird newlines around here

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala [hunk 3]
```diff
@@ -74,28 +74,49 @@ class UIProcessValidator(
...
+
+          val resolvedScenarioWithoutDisabledNodes =
+            fragmentResolver.resolveFragments(canonical.withoutDisabledNodes, processingType)
+
+          resolvedScenarioWithoutDisabledNodes match {
+            case Invalid(e)                          => formatErrors(e)
```
Comment (wrzontek) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala:106]:
```suggestion
            case Invalid(fragmentResolutionErrors)                          => formatErrors(fragmentResolutionErrors)
```
for consistency

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala (hunk 1 above)
Comment (mslabek) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala]: thanks for catching that. I'm pretty sure that there was some case where it was not the case, but cannot reproduce it now so it's probably just a mistake

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala (hunk 2 above)
Comment (mslabek) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala]: I didn't want to have a wide wall of text but it got weird. Do you think its ok now?

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala (hunk 3 above)
Comment (mslabek) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala]: done :+1:

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala [hunk 4]
```diff
@@ -74,28 +74,45 @@ class UIProcessValidator(
...
-              .withNodeResults(validated.typing.mapValuesNow(nodeInfoToResult))
-          case Invalid(e) => formatErrors(e)
+    val resolvedScenarioResult = fragmentResolver.resolveFragments(canonical, processingType)
+
+    // TODO: handle types when fragment resolution fails
+    val validationResult = resolvedScenarioResult match {
+      case Invalid(fragmentResolutionErrors) => formatErrors(fragmentResolutionErrors)
+      case Valid(scenario) =>
+        val validationResult = validateAndFormatResult(scenario)
+        val containsDisabledNodes =
```
Comment (wrzontek) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/validation/UIProcessValidator.scala:97]:
```suggestion
        val containsDisabledNodes = canonical.collectAllNodes.exists {
          case nodeData: Disableable if nodeData.isDisabled.contains(true) => true
          case _                                                           => false
        }
```

this seems cheaper than collecting all nodes twice

**Review by wrzontek — APPROVED** [review]

**Pull request merged**